dm/syncer/syncer.go

```diff
@@ -1497,69 +1541,53 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		flushCheckpoint = true
 		err = s.loadTableStructureFromDump(ctx)
 		if err != nil {
-			tctx.L().Warn("error happened when load table structure from dump files", zap.Error(err))
+			s.tctx.L().Warn("error happened when load table structure from dump files", zap.Error(err))
 			cleanDumpFile = false
 		}
 	} else {
 		cleanDumpFile = false
 	}
 
+	s.runWg.Add(1)
+	// start syncDML worker first because if there is no dml worker checkpointFlushWorker will blocked forever
+	go s.syncDML()
+	s.runWg.Add(1)
+	// start flush checkpoints worker. this worker mast start before s.flushCheckPoints()
+	go func() {
+		defer s.runWg.Done()
+		// also need to use a ctx different s.runCtx, checkpointFlushWorker worker will closed in the first defer
```

Comment:
```suggestion
		// also need to use a ctx different s.runCtx, checkpointFlushWorker worker will be closed in the first defer
```

many passive voice errors